### PR TITLE
Feature/sg 132 models convert

### PR DESCRIPTION
**Pull request by shaydeci**
- Added models.convert_to_onnx.
- Added conversion_params + convert_recipe example.
- In order to verify the conversion script works- I also added ontop of the train_from_recipe (i.e gpu test) integration test a call to convert_recipe, then in the recipe tests I'm verifying the onnx file does exist.
- Smal fix in cifar10 yaml (dient have multi_gpu mode).
- Standarize transform (because we don't have ToTensor..it works on np arrays).

### CODE REVIEW

File: src/super_gradients/training/models/conversion.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    :param post_process: torch.nn.Module, postprocessing pipeline, will be resolved by TransformsFactory()
+    :param prep_model_for_conversion_kwargs: dict, for SgModules- args to be passed to model.prep_model_for_conversion
+     prior to torch.onnx.export call.
+    :param torch_onnx_export_kwargs: kwargs (EXCLUDING: FIRST 3 KWARGS- MODEL, F, ARGS). to be unpacked in torch.onnx.export call
+
+    :return: out_path
+    """
+    torch_onnx_export_kwargs = torch_onnx_export_kwargs or dict()
+    prep_model_for_conversion_kwargs = prep_model_for_conversion_kwargs or dict()
+    onnx_input = torch.Tensor(np.zeros([1, *input_shape]))
```
Comment (ofrimasad) [inline src/super_gradients/training/models/conversion.py:73]: check that directory exists

**Review by ofrimasad — COMMENTED** [review]
my comments inline

File: src/super_gradients/common/object_names.py [hunk 2]
```diff
@@ -56,6 +56,7 @@ class Transforms:
     RandAugmentTransform = "RandAugmentTransform"
     Lighting = "Lighting"
     RandomErase = "RandomErase"
+    Standarize = "Standarize"
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply src/super_gradients/common/object_names.py]: fixed in https://github.com/Deci-AI/super-gradients/pull/598/commits/3386938c143cd7020663d84a325651406d283027

File: src/super_gradients/training/transforms/__init__.py [hunk 3]
```diff
@@ -26,6 +27,7 @@
     "DetectionPaddedRescale",
     "DetectionTargetsFormatTransform",
     "imported_albumentations_failure",
+    "Standarize",
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply src/super_gradients/training/transforms/__init__.py]: fixed in https://github.com/Deci-AI/super-gradients/pull/598/commits/3386938c143cd7020663d84a325651406d283027

File: src/super_gradients/training/transforms/transforms.py [hunk 4]
```diff
@@ -1104,3 +1105,20 @@ def rescale_and_pad_to_size(img, input_size, swap=(2, 0, 1), pad_val=114):
     padded_img = padded_img.transpose(swap)
     padded_img = np.ascontiguousarray(padded_img, dtype=np.float32)
     return padded_img, r
+
+
+class Standarize(torch.nn.Module):
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply src/super_gradients/training/transforms/transforms.py]: fixed in https://github.com/Deci-AI/super-gradients/pull/598/commits/3386938c143cd7020663d84a325651406d283027

File: src/super_gradients/recipes/conversion_params/default_conversion_params.yaml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+ckpt_root_dir: # The checkpoint root directory, s.t  ckpt_root_dir/experiment_name/ckpt_name resides.
+                # Can be ignored if the checkpoints directory is the default (i.e path to checkpoints module from contents root), or when checkpoint_path is given
+
+ckpt_name: ckpt_best.pth # Name of the checkpoint to export ("ckpt_latest.pth", "average_model.pth" or "ckpt_best.pth" for instance).
+checkpoint_path:
+strict_load: no_key_matching # See super_gradients/common/data_types/enum/strict_load.py
+
+# CONVERSION RELATED PARAMS
+out_path: # str, Destination path for the .onnx file. When None- will be set to the checkpoint_path.replace(".ckpt",".onnx").
+input_shape: # input shape, not including batch_size (i.e (3, 224, 224))
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply src/super_gradients/recipes/conversion_params/default_conversion_params.yaml:11]: Not gonna enforce passing it (also ??? us what you meant, and is equivalent to leaving empty).

File: src/super_gradients/training/models/conversion.py (hunk 1 above)
Comment (shaydeci) [reply src/super_gradients/training/models/conversion.py:77]: Added in https://github.com/Deci-AI/super-gradients/pull/598/commits/4a4007e91b841275d44f0a2fb879918b1a03c912

File: src/super_gradients/recipes/conversion_params/cifar10_conversion_params.yaml [hunk 6]
```diff
@@ -0,0 +1,36 @@
+# Example conversion parameters, to be used with super_gradients/examples/convert_recipe_example/convert_recipe_example.py
+# Important: must train cifar10_resnet using train_from_recipe beforehand. (see super_gradients/recipes/cifar10_resnet.yaml for complete instructions)
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply src/super_gradients/recipes/conversion_params/cifar10_conversion_params.yaml]: changed in https://github.com/Deci-AI/super-gradients/pull/598/commits/693f7387a0e477fc30d9d509bfa6009df41a369a

File: src/super_gradients/recipes/conversion_params/cifar10_conversion_params.yaml [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+#     python convert_recipe_example.py --config-name=cifar10_conversion_params checkpoint_path=YOUR_CHECKPOINT_PATH
+defaults:
+  - default_conversion_params
+  - _self_
+
+experiment_name: resnet18_cifar # The experiment name used to train the model (optional- ignored when checkpoint_path is given)
+
+# CONVERSION RELATED PARAMS
+out_path: # str, Destination path for the .onnx file. When None- out_path will be the resolved checkpoint path replacing .ckpt suffix with .onnx.
+input_shape: # input shape, not including batch_size (i.e (3, 224, 224))
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply src/super_gradients/recipes/conversion_params/cifar10_conversion_params.yaml]: changed in https://github.com/Deci-AI/super-gradients/pull/598/commits/693f7387a0e477fc30d9d509bfa6009df41a369a

File: src/super_gradients/recipes/conversion_params/cifar10_conversion_params.yaml [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+  - 3
+  - 32
+  - 32
+pre_process: # Preprocessing pipeline, will be resolved by TransformsFactory(), optional.
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply src/super_gradients/recipes/conversion_params/cifar10_conversion_params.yaml]: added in https://github.com/Deci-AI/super-gradients/pull/598/commits/693f7387a0e477fc30d9d509bfa6009df41a369a

File: src/super_gradients/recipes/conversion_params/default_conversion_params.yaml [hunk 9]
```diff
@@ -0,0 +1,15 @@
+experiment_name: # The experiment name used to train the model (optional- ignored when checkpoint_path is given)
+ckpt_root_dir: # The checkpoint root directory, s.t  ckpt_root_dir/experiment_name/ckpt_name resides.
+                # Can be ignored if the checkpoints directory is the default (i.e path to checkpoints module from contents root), or when checkpoint_path is given
+
+ckpt_name: ckpt_best.pth # Name of the checkpoint to export ("ckpt_latest.pth", "average_model.pth" or "ckpt_best.pth" for instance).
+checkpoint_path:
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply src/super_gradients/recipes/conversion_params/default_conversion_params.yaml:5]: added explanation in https://github.com/Deci-AI/super-gradients/pull/598/commits/693f7387a0e477fc30d9d509bfa6009df41a369a

File: src/super_gradients/recipes/conversion_params/default_conversion_params.yaml [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+strict_load: no_key_matching # See super_gradients/common/data_types/enum/strict_load.py
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply src/super_gradients/recipes/conversion_params/default_conversion_params.yaml]: added in https://github.com/Deci-AI/super-gradients/pull/598/commits/693f7387a0e477fc30d9d509bfa6009df41a369a

File: src/super_gradients/training/models/conversion.py [hunk 11]
```diff
@@ -0,0 +1,126 @@
+from pathlib import Path
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply src/super_gradients/training/models/conversion.py:1]: I like the logic of keeping it under models module.
We already talked about taking models level up, so lets leave it here.

**Review by ofrimasad — DISMISSED** [review]
LGTM

**Review by ofrimasad — APPROVED** [review]
LGTM
